app/storage/encrypted_questionnaire_storage.py

```diff
@@ -18,28 +20,35 @@ def __init__(self, user_id, user_ik, pepper):
         self.encrypter = StorageEncryption(user_id, user_ik, pepper)
 
     def add_or_update(self, data, version):
-        encrypted_data = self.encrypter.encrypt_data(data)
-        encrypted_data_json = json.dumps({'data': encrypted_data})
+        compressed_data = snappy.compress(data)
+        encrypted_data = self.encrypter.encrypt_data(compressed_data)
         questionnaire_state = self._find_questionnaire_state()
         if questionnaire_state:
             logger.debug('updating questionnaire data', user_id=self._user_id)
-            questionnaire_state.state_data = encrypted_data_json
+            questionnaire_state.state_data = encrypted_data
             questionnaire_state.version = version
         else:
             logger.debug('creating questionnaire data', user_id=self._user_id)
-            questionnaire_state = QuestionnaireState(self._user_id, encrypted_data_json, version)
+            questionnaire_state = QuestionnaireState(self._user_id, encrypted_data, version)
 
         data_access.put(questionnaire_state)
 
     def get_user_data(self):
         questionnaire_state = self._find_questionnaire_state()
-        if questionnaire_state:
-            data = json.loads(questionnaire_state.state_data)
+        if questionnaire_state and questionnaire_state.state_data:
             version = questionnaire_state.version or 0
 
-            if 'data' in data:
-                decrypted_data = self.encrypter.decrypt_data(data['data'])
-                return decrypted_data, version
+            if questionnaire_state.version < 3:
```

Comment:
Should we use version as above here? I guess that `or 0` was there from the original upgrade?